### DISCUSSION TITLE
Need help implementing Routed Events (for User Control)

### DISCUSSION THREAD

**Discussion by BertrandDeSaintBezier** [unoplatform/uno#13353] [Q&A] [answered]
The [Uno documentation](https://platform.uno/docs/articles/features/routed-events.html) states that [Routed Events](https://learn.microsoft.com/en-us/dotnet/desktop/wpf/events/routed-events-overview?view=netdesktop-7.0) for things like Tunneling, Bubbling, etc. are supported by Uno Platform, but I can't seem to get this piece of code to compile.
 
**Sample Code**
```xaml
<StackPanel Button.Click="TestButton_Click">
    <Button Name="Yes Button">Yes</Button>
    <Button Name="No Button">No</Button>
    <Button Name="Cancel Button">Cancel</Button>
</StackPanel>
```

I'm still just getting my head wrapped around the concept of Bubbling, so I'm probably missing something. What I'm eventually trying to acheieve is to catch an event that is Bubbling from a User Control :

**User Control**
```xaml
<Grid>
    <ListView ItemsSource="{x:Bind Files}" 
              SelectionChanged="{x:Bind ListView_SelectionChanged}">
        <ListView.ItemTemplate>
            ...
        </ListView.ItemTemplate>
    </ListView>
</Grid>
```

**Parent Element**
```xaml
<Grid HorizontalAlignment="Left" 
      VerticalAlignment="Top"
      ListView.SelectionChanged="{x:Bind ViewModel.HandleUserControlSelectionChanged}">
    <my:CustomListViewUserControl/>
</Grid>
``` 
I know I'm probably going about this the wrong way, because my User Control isn't a `ListView` per-se, so my guess is that the parent isn't going to catch it with `ListView.SelectionChanged`. My first priority though is confirming that what I'm trying to do isn't completely insane.

Alternatively if this comes off as a giant code smell to someone, please let me know, as I'd rather rethink the code architecture than build bad habits.

**Comment by MarkBrainfire** [comment]
Hi Bertrand,
I'll help you.  What are you trying to do?

**Reply by BertrandDeSaintBezier** [reply]
Hi Mark, sorry for the late reply. I actually ended up finding the solution to my problem shortly after, which is the same conclusion as the one in the answer post. 

Thanks for writing though !

**Comment by jeromelaban** [comment, marked as answer]
@BertrandDeSaintBezier you're looking at the WPF documentation, and event bubbling is only supported in WinUI for very specific events. See [this documentation](https://learn.microsoft.com/en-us/windows/uwp/xaml-platform/events-and-routed-events-overview) on the topic for UWP/WinUI.

In your sample, you'll need to explicitly attached the event on the `ListView` control for the event to be raised properly.

**Reply by BertrandDeSaintBezier** [reply]
Hi Jerome, sorry for the late reply. 

I ended up coming to the same conclusion shortly after. I don't know why I was trying to overcomplicate things. I guess tunnel vision will get the best of you sometimes 😉

Also, thanks for clearing up the fact that support is limited to only a couple of specific events !
